staging/src/k8s.io/apiserver/pkg/util/shufflesharding/shufflesharding_test.go

```diff
@@ -272,33 +275,73 @@ func TestShuffleAndDealToSliceWithValidation(t *testing.T) {
 
 func BenchmarkShuffleAndDealToSlice(b *testing.B) {
 	hashValueBase := math.MaxUint64 / uint64(b.N)
-	deckSize, handSize := int32(512), int32(8)
+	deckSize, handSize := 512, 8
 	for i := 0; i < b.N; i++ {
 		_ = ShuffleAndDealToSlice(hashValueBase*uint64(i), deckSize, handSize)
 	}
 }
 
+// ff computes the falling factorial `n!/(n-m)!` and requires n to be
+// positive and m to be in the range [0, m] and requires the answer to
+// fit in an int
+func ff(n, m int) int {
```

Comment:
does it make sense to validate the params to meet the conditions as described in the comment above?

Reply:
I think not, since this is not a public function; it only needs to be good enough for local usage.